### PR TITLE
(fix) update to new PublishTestResults version. Old version had path …

### PR DESCRIPTION
**Pull request by anthonyvercolano**
…bug.

### CODE REVIEW

File: vsts/node-nightly-df.yaml [hunk 1]
```diff
@@ -41,8 +39,12 @@ phases:
       IOT_PROVISIONING_ROOT_CERT: $(LINUX-IOT-PROVISIONING-ROOT-CERT)
       IOT_PROVISIONING_ROOT_CERT_KEY: $(LINUX-IOT-PROVISIONING-ROOT-CERT-KEY)
       IOT_PROVISIONING_SERVICE_CONNECTION_STRING: $(LINUX-IOT-PROVISIONING-SERVICE-CONNECTION-STRING)
+      DPS_CONN_STRING_INVALID_CERT: $(DPS-CONN-STRING-INVALID-CERT)
+      DPS_GLOBAL_DEVICE_ENDPOINT_INVALID_CERT: $(DPS-GLOBAL-DEVICE-ENDPOINT-INVALID-CERT)
+      IOTHUB_CONN_STRING_INVALID_CERT: $(IOTHUB-CONN-STRING-INVALID-CERT)
+      IOTHUB_DEVICE_CONN_STRING_INVALID_CERT: $(IOTHUB-DEVICE-CONN-STRING-INVALID-CERT)
 
-  - task: PublishTestResults@1
+  - task: PublishTestResults@2
```
Comment (pierreca) [inline vsts/node-nightly-df.yaml]: v1 used to work fine, and v2 had problems with glob patterns.. what changed that it doesn't work anymore?

Comment (anthonyvercolano) [reply vsts/node-nightly-df.yaml]: V1 was getting a path error.  Saying it was too long.  This was bogus.  The xml file paths and names  were no longer than before.  What had changed was that there was one additional xml from the server_validation test?  Going to v2 fixed it for me.

Comment (anthonyvercolano) [reply vsts/node-nightly-df.yaml]: This path issue had been noted as an issue for v1.

File: e2etests/test/device_acknowledge_tests.js [hunk 2]
```diff
@@ -61,19 +90,24 @@ function device_acknowledgment_tests (deviceTransport, createDeviceMethod) {
     });
 
     it('Service sends 1 C2D message and it is re-sent until completed', function (done) {
+      this.timeout(15000);
```
Comment (pierreca) [inline e2etests/test/device_acknowledge_tests.js]: >15000 [](start = 19, length = 5)

that's very short we usually go for 60 seconds

File: e2etests/test/device_acknowledge_tests.js [hunk 3]
```diff
@@ -17,6 +17,36 @@ var deviceHttp = require('azure-iot-device-http');
...
+  }
+  this.doneYet[participant] = true;
+  this.everybodyDone = true;
+  Object.keys(this.doneYet).forEach(function(aParticipant) {
+    this.everybodyDone = this.everybodyDone && this.doneYet[aParticipant];
+  }.bind(this));
+  if (this.everybodyDone) {
+    return this.done();
+  }
+};
```
Comment (pierreca) [inline e2etests/test/device_acknowledge_tests.js]: that is not self-documenting at all. please consider naming these appropriately or adding doc comments.

File: e2etests/test/device_acknowledge_tests.js (hunk 2 above)
Comment (anthonyvercolano) [reply e2etests/test/device_acknowledge_tests.js]: I've done thousands of runs on these tests.  This timeout was triggered except for when there was "loss" of a message and it was never going to come in.  Note that this this is on the individual tests.

**Pull request merged**